File: DS/LinkedList/DeleteNodeAtPosition.py
"""
Delete Node At a specific Position
If Position is 0 then return the nextnode. [So, setting next node to head Position means delete head node.]
else need to set nextnode in recursive with decrementing Position.
"""

class Node:
    def __init__ (self,data,next):
        self.data= data
        self.next = nextnode

class LinkedList:
    def __init__ (self):
        self.currnode = None
    
    def addNode(self, data):
        node = Node(data,self.currnode)
        self.currnode = node
    
    def Delete(head, position):        
        if position == 0:
            return head.next
        head.next = Delete(head.next, position - 1)
        return head

File: DS/LinkedList/DeleteNodeAtPosition.cs
//Delete Node At a specific Position

using System;

namespace ConsoleApplication1
{
    class Nodes
    {
        public int data;
        public Nodes nextnode;
        public Nodes(int _data, Nodes _nextnode)
        {
            this.data = _data;
            this.nextnode = _nextnode;
        }
    }
    class LinkedList
    {
        Nodes currnode;

        public LinkedList()
        {
            this.currnode = null;
        }

        void Add(int data)
        {
            Nodes _nodes = new Nodes(data, this.currnode);
            this.currnode = _nodes;
        }

        Nodes DeleteNodeAtPos(Nodes head,int Pos)
        {
            if(Pos == 0)
            {
                return head.nextnode;
            }

            head.nextnode = DeleteNodeAtPos(head.nextnode,Pos-1);
            return head;
        }
    }
}
